lib/navigation/preprocess_encoders.hpp

```diff
@@ -1,27 +1,104 @@
 #pragma once
+#include "consts.hpp"
 
+#include <cmath>
+
+#include <algorithm>
 #include <array>
 #include <cstdint>
+#include <optional>
 
+#include "core/logger.hpp"
 #include "core/types.hpp"
 
 namespace hyped::navigation {
+
 class EncodersPreprocessor {
  public:
-  EncodersPreprocessor();
+  EncodersPreprocessor(core::ILogger &logger);
 
-  core::EncoderData processData(const core::EncoderData encoder_data);
+  /**
+   * @brief process encoder data to deal with outliers and unreliable sensors
+   *
+   * @param encoder_data
+   * @return cleaned and reliable encoder data
+   */
+  std::optional<core::EncoderData> processData(const core::EncoderData &encoder_data);
 
  private:
-  core::EncoderData detectOutliers(const core::EncoderData encoder_data);
+  struct Statistics {
+    core::Float median;
+    core::Float upper_bound;
+    core::Float lower_bound;
+  };
+
+  /**
+   * @brief calculate the upper and lower bounds as well as the median for
+   * the input data array
+   *
+   * @param encoder_data
+   * @return std::optional<Statistics>
+   */
```

Comment:
remove, no valuable information is contained in this

just from reading `struct Statistics { ... }` and the signature all of these things can be inferred